airbyte-webapp/src/components/connection/CatalogTree/streamConfigHelpers/streamConfigHelpers.ts

```diff
@@ -6,17 +6,18 @@ import { AirbyteStreamConfiguration, SelectedFieldInfo } from "core/request/Airb
  * Merges arrays of SelectedFieldInfo, ensuring there are no duplicates
  */
 export function mergeFieldPathArrays(...args: SelectedFieldInfo[][]): SelectedFieldInfo[] {
-  const set = new Set<string[]>();
+  const set = new Set<string>();
 
   args.forEach((array) =>
     array.forEach((selectedFieldInfo) => {
       if (selectedFieldInfo.fieldPath) {
-        set.add(selectedFieldInfo.fieldPath);
+        const key = JSON.stringify(selectedFieldInfo.fieldPath);
+        set.add(key);
```

Comment:
This converts the `fieldPath` (e.g. `['path', 'to', 'field']`) into a string (e.g. `"[\"path\",\"to\",\"field\"]"`) so that we can add it as a unique value to the set. It gets converted back into an array before returning.

This conversion between array > string > array might seem expensive, but it allows us use a `Set` to avoid duplicates when merging, which avoids a lot of nested comparisons of arrays.

Reply:
I think that's a reasonable tradeoff.  It sounds like it would still be more efficient for a large sync catalog or a sync catalog with a lot of changes than what this was doing before, correct?

Reply:
Yes, at least as efficient, and in the best case more efficient 👍